cmd/atlas/atlas.go

```diff
@@ -134,7 +136,11 @@ func mongoCLIConfigFilePath() (configPath string, err error) {
 
 func main() {
 	cobra.EnableCommandSorting = false
-	cobra.OnInitialize(createConfigFromMongoCLIConfig, initConfig)
+
+	createConfigFromMongoCLIConfig()
+	loadConfig()
+
+	cobra.OnInitialize(initProfile)
```

Comment:
can you please provide some details about the change? the new code seems to do exactly what the previous one was doing, is this affecting setting up telemetry somehow?

Reply:
well the key here is to load configuration outside of the `OnInitialize` block, `OnInitialize` is only triggered if all flags are valid, if you trigger command `atlas clusters ls --blah` for instance configuration would never be loaded, hence `telemetry_enabled` would always be `false`.

Reply:
thanks for clarifying

Reply:
also setting the profile depends on `profile` flag, so that bit has to stay on `OnInitialize` block